Source/utils/sdl_ptrs.h

```diff
@@ -11,28 +11,18 @@
 namespace devilution {
 
 /**
- * @brief Deletes the SDL surface using `SDL_FreeSurface`.
- */
-struct SDLSurfaceDeleter {
-	void operator()(SDL_Surface *surface) const
-	{
-		SDL_FreeSurface(surface);
-	}
-};
-
-using SDLSurfaceUniquePtr = std::unique_ptr<SDL_Surface, SDLSurfaceDeleter>;

```

Comment:
I much prefer separate names like before.

If you put it all in the same class, it couples things somewhat unnecessarily (always have to include the whole `<SDL.h>`) and it is unclear whether the correct deleter has been implemented without looking at the source.

Reply:
I prefer the new approach; the old one was confusing. Looking at the SDL docs, there are only a handful  of SDL_Free* functions, and we can support them exhaustively. The API isn't going to change until SDL3.

As far as having to include all of SDL, you still need it if you want the asserts from the old SDLFreeDeleter.